src/uct/ib/mlx5/dc/dc_mlx5.c

```diff
@@ -267,14 +270,42 @@ static void uct_dc_mlx5_iface_progress_enable(uct_iface_h tl_iface, unsigned fla
     uct_base_iface_progress_enable_cb(&iface->super.super, iface->progress, flags);
 }
 
+static void uct_dc_mlx5_asan_cleanup_old_dcis_buffer(uct_dc_mlx5_iface_t *iface)
+{
+#ifdef __SANITIZE_ADDRESS__
+    if (iface->old_dcis_buffer != NULL) {
+        ASAN_UNPOISON_MEMORY_REGION(iface->old_dcis_buffer,
+                                    iface->old_dcis_buffer_size);
+        ucs_free(iface->old_dcis_buffer);
+    }
+#endif
+}
+
+void uct_dc_mlx5_asan_relocate_dcis_array(uct_dc_mlx5_iface_t *iface)
+{
+#ifdef __SANITIZE_ADDRESS__
+    size_t buffer_size         = sizeof(uct_dc_dci_t) *
+                                 ucs_array_capacity(&iface->tx.dcis);
+    size_t num_dcis            = ucs_array_length(&iface->tx.dcis);
+    uct_dc_dci_t *old_buffer_p = ucs_array_begin(&iface->tx.dcis);
```

Comment:
Seems both `old` and `new` are `buffer`, why the 1st with `_p` and the second w/o?